### ISSUE TITLE
Improve Prometheus Compatibility: Respond to requests at /prometheus/-/healthy

### ISSUE THREAD

**Issue by jem-Solytica** [grafana/mimir#7399]
#### Is your feature request related to a problem? Please describe.

There is a Prometheus compatibility problem where an application checks that the Prometheus endpoint is healthy before making numerous requests for metrics by calling ```http://<prometheus>:<port>/-/healthy```. When pointing at Mimir, this endpoint returns a 404.

MinIO dashboard metrics are an example where in-dashboard metrics fail to load because the initial check fails when calling Mimir. (See this issue: https://github.com/minio/minio/issues/17996)

#### Describe the solution you'd like

Simply return ```Prometheus Server is Healthy.``` in response to a ```GET``` to ```http://<mimir>:<port>/prometheus/-/healthy```

#### Describe alternatives you've considered

Tried to find a workaround with the target application. Found none. 

#### Additional context

None.

**Comment by 56quarters** [comment]
Hey, Sorry to send you back to Minio again but I don't believe this is a Mimir issue. I'll explain.

`/-/healthy` isn't documented as part of the [Prometheus API](https://prometheus.io/docs/prometheus/latest/querying/api/) and so isn't implemented by Mimir or most other backends that support the Prometheus API (Cortex/Thanos/etc.). This endpoint also isn't a part of the [compatibility testing](https://promlabs.com/blog/2021/10/14/promql-vendor-compatibility-round-three/) that Prometheus does.

The `/-/healthy` endpoint is in fact specific to the [prometheus/prometheus](https://github.com/prometheus/prometheus) server. Testing that it exists doesn't test that a backend will support Prometheus queries, it only tests that a backend has that particular endpoint.

Additionally, a `/-/healthy` endpoint tells you nothing about if a distributed system is available. An individual component may by up, but the system as a whole may not be available.

If Minio wants to check that a PromQL backend is available, a better check would be to perform a no-op query like `vector(0)`.

Example:
```
curl -v 'http://mimir/prometheus/api/v1/query?query=vector(0)'
```

**Comment by jem-Solytica** [comment]
Thanks for the quick response. I've opened a new issue on the MinIO side.

**Comment by chemist777** [comment]
As a workaround, I solved this issue by using a tiny nginx server config as a proxy in front of Mimir.

In MinIO: `MINIO_PROMETHEUS_URL: http://127.0.0.1/prometheus`.

In Nginx:
```bash
server {
    listen 127.0.0.1:80;

    # Emulate the original Prometheus server
    location /prometheus/-/healthy {
        return 200 "Healthy\n";
        add_header Content-Type text/plain;
    }

    # Proxy all other Prometheus requests to Mimir
    location /prometheus/ {
        resolver 8.8.8.8 8.8.4.4;
        # "set" is needed to re-resolve DNS upon each request
        # It is needed because nginx caches DNS infinitely
        set $backend_url "https://some-mimir-server.com";
        proxy_pass $backend_url$request_uri;
    }
}
```

**Comment by johnlaur-boc** [comment]
Achieving this workaround in the `mimir-distributed` helm chart:

```yaml
gateway:
  nginx:
    config:
      serverSnippet: |
        # Prometheus compatibility shims won't be implemented (grafana/mimir#7399)
        location = /prometheus/-/healthy {
          return 200 "Prometheus Server is Healthy.\n";
        }
        location = /prometheus/-/ready {
          return 200 "Prometheus Server is Ready.\n";
        }
```
